StereoKitC/tools/virtual_keyboard.cpp

```diff
@@ -53,12 +53,14 @@ uint64_t                  keyboard_go_toggle        = 0;
 int32_t                   keyboard_visit_return_idx = -1;
 text_context_             keyboard_text_context     = text_context_text;
 
-array_t<keylayout_info_t> keyboard_layers           = {};
-int32_t                   keyboard_active_root      = 0;
-int32_t                   keyboard_active_idx       = 0;
-int32_t                   keyboard_ctx_root_text    = 0;
-int32_t                   keyboard_ctx_root_number  = 0;
-int32_t                   keyboard_ctx_root_uri     = 0;
+hashmap_t<text_context_, array_t<keylayout_info_t>> keyboards = {};
```

Comment:
Maybe we can shift `text_context_` to be just a consecutive 0,1,2,3.., and then use that to index a normal array of `array_t`? `hashmap_t` is okay, but reallllly feels like overkill here.